### PR TITLE
Add async File.ReadAll*, File.AppendAll* and File.WriteAll*

### PR DESCRIPTION
**Pull request by JonHanna**
See dotnet/corefx#11220 and dotnet/corefx#14557

### CODE REVIEW

File: src/mscorlib/src/System/IO/File.cs [hunk 1]
```diff
@@ -821,5 +823,323 @@ internal static int FillAttributeInfo(String path, ref Win32Native.WIN32_FILE_AT
...
+        {
+            Contract.Requires(writer != null);
+            Contract.Requires(contents != null);
+
+            using (writer)
+            {
+                foreach (String line in contents)
+                {
+                    cancel.ThrowIfCancellationRequested();
+                    await (line == null ? writer.WriteLineAsync() : writer.WriteLineAsync(line)).ConfigureAwait(false);
```
Comment (JonHanna) [inline src/mscorlib/src/System/IO/File.cs]: Having a null-check where there is none in the non-async version is a workaround for #8661

**Comment by JonHanna** [comment]
Test Windows_NT x64 Release Priority 1 Build and Test

https://ci.dot.net/job/dotnet_coreclr/job/master/job/x64_release_windows_nt_pri1_prtest/1178/testReport/junit/GC/_Scenarios_DoublinkList_dlbigleakthd_dlbigleakthd_/_Scenarios_DoublinkList_dlbigleakthd_dlbigleakthd_cmd/ seems unrelated to this?

**Comment by JonHanna** [comment]
Test Ubuntu x64 Checked Build and Test

**Comment by jkotas** [comment]
I do not think this is needed in CoreCLR. The shipping `class File` is in corefx.

@JeremyKuhne has been working on cleaning up the unused parts of File I/O from CoreCLR.

**Pull request closed without merging**